lib/patches/ancestry_patch.rb

```diff
@@ -8,16 +8,6 @@ def update_descendants_with_new_ancestry
       end
     end
   end
-
-  def parent=(parent)
-    super
-    clear_memoized_instance_variables
-  end
-
-  def parent_id=(parent)
-    super
-    clear_memoized_instance_variables
-  end
```

Comment:
I feel like we shouldn't remove this until we remove with_relationship_type entirely.  Either that, or maybe we should rename with_relationship_type since the semantics are kind of changing now.